### PR TITLE
Added tree view component

### PR DESCRIPTION
**Pull request by Johanh-od**

### CODE REVIEW

File: core/ui/prime/components/src/lib/input-base/input-base.component.ts [hunk 1]
```diff
@@ -1,85 +1,81 @@
+/* eslint-disable @typescript-eslint/no-empty-function */
```
Comment (bart-vandervliet) [inline core/ui/prime/components/src/lib/input-base/input-base.component.ts:1]: Why has this file so many changes? Spaces to tab?

File: core/ui/prime/components/src/lib/tree/tree.component.ts [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+	private selectedNodes: TreeNode<ITreeNode>[] = [];
+	private selectedNode: TreeNode<ITreeNode> | undefined;
+	private _singleSelectionMode: "single" | "multiple" | "Checkbox" = "single";
+
+	@Input()
+	public filter = false;
+	@Input()
+	public filterMode: "lenient" | "strict" = "lenient";
+	@Input()
+	public selectionMode: "single" | "multiple" | "Checkbox" = this._singleSelectionMode;
```
Comment (bart-vandervliet) [inline core/ui/prime/components/src/lib/tree/tree.component.ts:27]: Lowercase `checkbox`.

**Review by bart-vandervliet — APPROVED** [review]

File: core/ui/prime/components/src/lib/input-base/input-base.component.ts (hunk 1 above)
Comment (Johanh-od) [reply core/ui/prime/components/src/lib/input-base/input-base.component.ts:1]: Also structured the code a bit and do some formatting

File: core/ui/prime/components/src/lib/tree/tree.component.ts (hunk 2 above)
Comment (Johanh-od) [reply core/ui/prime/components/src/lib/tree/tree.component.ts:27]: Good one, will fix it

**Pull request merged**